rest/route/host_agent.go

```diff
@@ -525,6 +525,26 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 				"host_id":   currentHost.Id,
 			}))
 
+			if nextTask.IsStuckTask() {
```

Comment:
I'm pretty sure if we want to check for the issue described in the ticket (a host being assigned a task, and then asking for that task over and over again), we have to check it [here in sendBackRunningTask](https://github.com/evergreen-ci/evergreen/blob/8fe05b3b8f9e14487101ac82e3f308e959b8b2d7/rest/route/host_agent.go#L182-L1846) instead of in `assignNextAvailableTask`. If a host has already been assigned a task by the app server, its RunningTask will be set, so that case will just hand the already-assigned task back to the host again, meaning it'll never reach `assignNextAvailableTask` with that same task again.